### PR TITLE
daemon: do not use pointers to map

### PR DESCRIPTION
**Pull request by LK4D4**
Also, do not shadow err with :=

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -991,7 +989,7 @@ func (daemon *Daemon) Reload(config *Config) error {
 	}
 	if config.IsValueSet("live-restore") {
 		daemon.configStore.LiveRestoreEnabled = config.LiveRestoreEnabled
-		if err := daemon.containerdRemote.UpdateOptions(libcontainerd.WithLiveRestore(config.LiveRestoreEnabled)); err != nil {
+		if err = daemon.containerdRemote.UpdateOptions(libcontainerd.WithLiveRestore(config.LiveRestoreEnabled)); err != nil {
```
Comment (tonistiigi) [inline daemon/daemon.go]: We should probably make `err` a named return var instead.

Comment (LK4D4) [inline daemon/daemon.go]: Can do

File: daemon/daemon.go [hunk 2]
```diff
@@ -963,8 +963,11 @@ func (daemon *Daemon) initDiscovery(config *Config) error {
 // - Daemon live restore
 func (daemon *Daemon) Reload(config *Config) error {
 	var err error
-	// used to hold reloaded changes
-	attributes := map[string]string{}
+
+	daemon.configStore.reloadLock.Lock()
+	defer daemon.configStore.reloadLock.Unlock()
```
Comment (aaronlehmann) [inline daemon/daemon.go]: Reordering the locking means that `reloadLock` will be held through the call to `LogDaemonEventWithAttributes`, which can block for up to 100 ms. I'm not sure it matters, but it seems worth pointing out.

File: daemon/daemon.go (hunk 1 above)
Comment (LK4D4) [inline daemon/daemon.go]: Done!

File: daemon/daemon.go (hunk 2 above)
Comment (LK4D4) [inline daemon/daemon.go]: Hmm, yeah, I also see comment that `SystemInfo` holds that Lock, but it isn't, which is probably and error.

Comment (LK4D4) [inline daemon/daemon.go]: Oh, then we're ok, I will just move Unlock().

**Comment by LK4D4** [comment]
@tonistiigi @aaronlehmann Updated named var and mutex stuff.

**Comment by vieux** [comment]
LGTM

**Comment by aaronlehmann** [comment]
LGTM after comment nit is addressed

**Comment by tonistiigi** [comment]
LGTM

**Comment by LK4D4** [comment]
@aaronlehmann update comment, PTAL

**Pull request merged**